src/components/localization/localized-link.js

```diff
@@ -67,214 +47,188 @@ export const SharedLinkStyle = css`
-            <ExternalLink href={to} ref={ref} onClick={onClick} disabled={!has_mounted}>
-                {props.children}
-            </ExternalLink>
-        )
+    const is_index = to === `/`
+    const localized_url = getLocalizedUrl(path, is_index, to)
 
+    const path_to = is_default || is_non_localized ? to : localized_url
     let internal_to = path_to
 
     // remove trailing character only if it is a forward slash
-    if (props.has_no_end_slash && internal_to.charAt(internal_to.length - 1) == '/') {
-        internal_to = internal_to.substring(0, internal_to.length - 1)
-    }
+    internal_to = has_no_end_slash ? internal_to.replace(/\/$/, '') : internal_to
 
-    if (props.anchor) {
+    if (is_anchor) {
         return (
             <StyledAnchorLink
-                title={ariaLabel}
-                {...props}
+                title={aria_label}
                 to={internal_to}
-                ref={ref}
-                disabled={!has_mounted}
+                disabled={!mounted}
+                {...props}
             />
         )
     }
-
     return (
-        <StyledGatsbyLink
-            aria-label={ariaLabel}
-            target={target}
-            rel={rel}
-            className={className}
-            style={style}
-            to={internal_to}
-            ref={ref}
-            onClick={onClick}
-            disabled={!has_mounted}
-        >
-            {props.children}
+        <StyledGatsbyLink aria-label={aria_label} to={internal_to} disabled={!mounted} {...props}>
+            {children}
         </StyledGatsbyLink>
     )
-})
-
-LocalizedLink.displayName = 'LocalizedLink'
+}
 
-LocalizedLink.propTypes = {
-    anchor: PropTypes.bool,
-    ariaLabel: PropTypes.string,
+InternalLink.propTypes = {
+    aria_label: PropTypes.string,
     children: PropTypes.node,
-    className: PropTypes.string,
-    external: PropTypes.string,
-    external_link: PropTypes.bool,
     has_no_end_slash: PropTypes.bool,
-    is_affiliate_link: PropTypes.bool,
-    is_affiliate_sign_in_link: PropTypes.bool,
-    is_binary_link: PropTypes.bool,
-    is_blog_link: PropTypes.bool,
-    is_community_link: PropTypes.bool,
-    is_dbot_link: PropTypes.bool,
-    is_deriv_app_link: PropTypes.bool,
-    is_deriv_developer_link: PropTypes.bool,
+    is_anchor: PropTypes.bool,
+    locale: PropTypes.any,
+    mounted: PropTypes.bool,
+    to: PropTypes.string.isRequired,
+}
+
+const affiliate_links = ['affiliate_sign_in', 'affiliate_sign_up']
+const deriv_app_links = ['dbot', 'deriv_app', 'mt5']
+const deriv_other_products = ['binary', 'smart_trader']
+const deriv_social_platforms = ['blog', 'community', 'developers', 'zoho']
+
+const getURLFormat = (type, locale, to, affiliate_lang) => {
+    if (deriv_app_links.includes(type)) {
+        return getDerivAppLocalizedURL(localized_link_url[type], locale)
+    } else if (affiliate_links.includes(type)) {
+        return `${localized_link_url[type]}?lang=${affiliate_lang}`
```

Comment:
Do you think we should we call this `localized_link_url[type]` as a `getLocalizedLink` function?

Reply:
The `localized_link_url` is a constant with key value pair. I think it would be good to leave it like this for now. Refactoring it to function would need add an additional variable as placeholder and also an extra return statement. wdyt?